development/process.md

```diff
@@ -101,6 +102,93 @@ keep the amount of rework from the contributor to a minimum.
       website are updated.
     * Finally the release is announced.
 
+## Pull request and commit guidelines <a name="commit"></a>
+
+Contributions to Trino projects are managed as [collaboration with pull
+requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests).
+
+Pull requests are usually merged into `master` or `main` using the  [`rebase and
+merge`](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits)
+strategy.
+
+A typical pull request should strive to contain a single logical change, but not
+necessarily a single commit. Unrelated changes should generally be extracted
+into their own PRs.
+
+If a pull request contains a stack of more than one commit, then popping any
+number of commits from the top of the stack, should not break the PR, meaning
+that  every commit should build and pass all tests.
+
+Commit messages and history are important as well, because they are used by
+other developers to keep track of the motivation behind changes. Keep logical
+diffs grouped together in separate commits and order commits in a way that
+explains by itself the evolution of the change. Rewriting and reordering commits
+is a natural part of the review process.
+
+Mechanical changes like refactoring, renaming, removing duplication, extracting
+helper methods, static imports should be kept separated from logical and
+functional changes like adding a new feature or modifying code behaviour. This
+makes reviewing the code much easier and reduces the chance of introducing
+unintended changes in behavior.
+
+Whenever in doubt on splitting a change into a separate commit, ask yourself the
+following question: if all other work in the PR needs to be reverted after
+merging to master for some objective reason, such as a bug has been discovered,
+is it worth keeping that commit still in master.
+
+When writing a commit message, follow the seven rules of a great commit message:
+
+* Separate subject from body with a blank line.
+* Limit the subject line to 50 characters.
+* Capitalize the subject line
+* Do not end the subject line with a period.
+* Use the imperative mood in the subject line.
+* Wrap the body at 72 characters.
+* Use the body to explain what and why versus how.
```

Comment:
I think 50, 72 are obsolete these days - but whatever.

Reply:
this is quote from the page we link to .. 72 is better ..since a bit longer